### DISCUSSION TITLE
Support a 'Disabled' property on data sources

### DISCUSSION THREAD

**Discussion by dylan-asos** [grafana/grafana#65026] [Data Sources]
Support a simple true/false disabled property on data sources. 

Simply renaming or otherwise removing the a data source can have adverse impact, with panels defaulting to another data source and panel queries becoming hidden. 

Expected behaviour : 

  - On data source maintenance page, support a 'disabled' option
  - API calls to the proxy endpoint that use a disabled data source should return a response "Query execution failed, data source is in disabled state'
  - Error message should be written to logs, with the UID of the dashboard / panel that executed the disabled query so it can be easily located by grafana admins
  - On any panel that uses a disabled data source, at query execution time after receiving the API response, an error status indicator should be displayed on the panel 'Data source is currently disabled'
 
Disabled data sources could provide a number of possibilities

* Temporarily stop traffic to a target while the target is being maintained or otherwise shouldn't be receiving traffic
* Gracefully migrate users to other data sources. In larger deployments, with many dashboards using many data sources, you may need to deprecate data sources and migrate users to other options - this would support a scenario to help find usage, and allow users to update while keeping the UI+query easy to work with.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82751. Feel free to continue the discussion around this there. Thank you!
